dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -95,37 +95,80 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
     for {
       rValDbs <- rValueDAO.findAll()
       eventDbs <- eventDAO.findAll()
-      outcomes <- eventOutcomeDAO.findAll()
     } yield {
       val rValDbsByNonce = rValDbs.groupBy(_.nonce)
-      val outcomesByNonce = outcomes.groupBy(_.nonce)
-      eventDbs.map(db =>
-        Event(rValDbsByNonce(db.nonce).head, db, outcomesByNonce(db.nonce)))
+      eventDbs.map(db => Event(rValDbsByNonce(db.nonce).head, db))
     }
   }
 
   def getEvent(nonce: SchnorrNonce): Future[Option[Event]] = {
     for {
       rValDbOpt <- rValueDAO.read(nonce)
       eventDbOpt <- eventDAO.read(nonce)
-      outcomes <- eventOutcomeDAO.findByNonce(nonce)
     } yield {
       (rValDbOpt, eventDbOpt) match {
         case (Some(rValDb), Some(eventDb)) =>
-          Some(Event(rValDb, eventDb, outcomes))
+          Some(Event(rValDb, eventDb))
         case (None, None) | (Some(_), None) | (None, Some(_)) =>
           None
       }
     }
   }
 
-  def createNewEvent(
+  /** Creates an event over a range of numbers from start to stop (inclusive)
+    * increasing by the step.
+    *
+    * If the final step would be higher than stop, it will not go past it.
+    */
+  def createNewRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      start: Int,
+      stop: Int,
+      step: Int): Future[EventDb] =
+    createNewRangedEvent(eventName,
+                         maturationTime,
+                         Int32(start),
+                         Int32(stop),
+                         UInt16(step))
+
+  /** Creates an event over a range of numbers from start to stop (inclusive)
+    * increasing by the step.
+    *
+    * If the final step would be higher than stop, it will not go past it.
+    */
+  def createNewRangedEvent(
+      eventName: String,
+      maturationTime: Instant,
+      start: Int32,
+      stop: Int32,
+      step: UInt16): Future[EventDb] = {
+    require(stop >= start, s"Start ($start) must be after Stop ($stop)")
```

Comment:
Currently it is not enforced at all. I'm apprehensive of enforcing it anywhere because the oracle is a representation of the real world and things don't always go as expected. 

I can see us requiring that the maturation date be in the future, however, since nothing is enforced after that I don't really see the point. I think we should let the user set whatever time they want.

Reply:
Can you do DLCs based on past events? 

The only reason I can think of allowing for maturation times in the past is if you are transferring data from one database to another and you want to use this method to do it. I would argue you should use the underlying `DAO`s though.